sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -481,6 +481,16 @@ internal void EnsureState()
                 else
                 {
                     Resource = Constants.Sas.Resource.Directory;
+                    if (!Path.Equals("/", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
I think maybe we should add a unit test with a couple `TestCase()`s for directory depth calculation.